quality-assurance/core.md

```diff
@@ -0,0 +1,194 @@
+* Autofill is disabled on inputs where appropriate
+* Inputs have correct type property ( emails have email type, phone numbers have tel type)
+
+**Modals**
+* Modals are animated on both display and dismiss
+* Modals are used where appropriate over pages
+
+**Formatting**
+* Dates are formatted correctly for the appropriate audience.
+* Monetary values are formatted correctly for the appropriate audience
+* Phone numbers are formatted correctly for the appropriate audience
+
+**Pagination**
+* UI correctly indicates current page
+* UI shows at least 2 page tabs ahead and below the current page
+
+**Login &amp; Auth**
+* User can only access pages that require 
+* Authentication when they are logged in
+* Login &amp; Logout works as expected
+* Reset password works as expected
+
+**Registration**
+* Email addresses are verified &amp; unique (if applicable)
+* Passwords meet minimum standards (min 8 chars)
+* Passwords should not have maximum length restrictions (allow at least 255 characters).
+
+**Emails**
+* Emails look good on small form factor devices
+* Ensure correct subject &amp; sender name
+* Correct email is sent, information is conveyed appropriately
+* Emails are received in a timely manner
+* Emails come from the appropriate email (often the clients)
+* Links point to the correct address
+
+**Push Notifications**
+* Push notifications are received in a timely manner
+* They show correct &amp; relevant information
+
+**Permissions Roles**
+* Ensure permissions &amp; roles match up to possible actions (only admins can create users etc)
+
+### Whitebox
+**General**
+* Filenames must follow standard coding conventions (UsersController.php)
+* Classes must follow standard coding conventions (UsersController)
+* Variables should follow camelCase &amp; be descriptive (myName)
+* All long running operations (at your discretion) are queued
+* APIs follow REST architecture as closely as possible
+* Auth middleware is setup &amp; working correctly
+* Event system is used for method side effects (Audit trail etc)
+* Laravel Notifications are used to notify the user of system events (subscription expired, new friend request etc)
+* Sensitive values are either hashed or encrypted (when required by law or common sense)
+* Third party API integrations that can &amp; will be re-used across projects should be implemented as external packages, 
+assuming no existing open-source integration exists, to allow for independent testing and code reuse if appropriate.
+* Ensure third party API keys (e.g. SES keys) are not present within .env.example and other relevant files outside of the 
+production / development environments.
+* If Travis CI is being used, ensure third party API keys are input directly into Travis CI's website relevant to the 
+project and make sure "Display value in build log" is not checked.
+* Log viewers are hidden behind a protected route
+
+**Controllers**
+* Ensure each controller function does not go over 250 lines
```

Comment:
This should be the entire class and not just function, a single function should really never go over 50 lines.

Reply:
> This should be the entire class and not just function, a single function should really never go over 50 lines.

This is your document 🤕

Reply:
Just pointing out a typo.